### PR TITLE
Add instructions for making a release

### PR DESCRIPTION
**Pull request by nrc**
I hope I remembered everything that needs doing. r? @alexcrichton

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -28,3 +28,14 @@ Unless you explicitly state otherwise, any contribution intentionally
...
+
+## Making a release
+
+Producing the final release artifacts is a bit involved because of the way Rustup
+is distributed. The steps for a release are:
+
+* Update the version number in all Cargo.tomls
+* `cargo build` to update the lock files.
+* commit and tag (`git commit -a` and `git tag -a $VER_NUM -m $VER_NUM`)
+* push the branch and tag (`git push upstream master` and `git push upstream $VER_NUM`)
```
Comment (dwijnand) [inline CONTRIBUTING.md]: That's `git push --follow-tags`, by the way. 😉

**Review by dwijnand — COMMENTED** [review]
Excellent.  Thank you, Nick.

**Comment by alexcrichton** [comment]
I think after the lock file the steps may want to be updated to:

* Push to the `stable` branch, wait for CI to pass
* Ping the release team to perform the actual release, instructions in [`ci/sync-dist.py`](https://github.com/rust-lang/rustup.rs/blob/master/ci/sync-dist.py). Note that some manual testing happens here too
* After release is done, do the tags and push the tags.

Other than that though looks good to me!

**Comment by kinnison** [comment]
I wonder if we ought to be syncing the version string in `rustup-init.sh` too as part of the release process?

**Comment by nrc** [comment]
rustup-init version string is at https://github.com/rust-lang/rustup.rs/blob/master/rustup-init.sh#L25

**Comment by kinnison** [comment]
The checklist I made as I prepared the release was:

```markdown
1. Update all `Cargo.toml` to have the new version
   (optionally make a commit)
2. Run `cargo build` and review `Cargo.lock` changes
   if all looks well, make a commit
3. Update `rustup-init.sh` version to match the commit
   details for `Cargo.lock`
4. Push this to the stable branch (git push origin HEAD:stable)
5. Wait for green CI
6. Ping someone in the release team to perform the actual release.
   They can find instructions in `ci/sync-dist.py`
   Note: Some manual testing occurs here, so hopefully they'll catch
   anything egregious in which case abort the change and roll back.
7. Once the official release has happened, prepare and push a tag
   of that commit, and also push the content to master
   * `git tag -as $VER_NUM -m $VER_NUM`  (optionally without -s if not GPG
     signing the tag)
   * `git push origin HEAD:master`
   * `git push origin $VER_NUM`
```

**Comment by kinnison** [comment]
We might need to also add to that a bullet point about manual testing of `rustup-init.sh` too

**Comment by kinnison** [comment]
Whoo-weee, and perhaps also a note about checking we've not broken `rls-vscode` too -- gosh lots of things run rustup and are delicate with parsing our output.  More grist for the "We should have a plumbing API" mill I guess :D

**Pull request merged**